assignment2/BinaryTree.java

```diff
@@ -0,0 +1,198 @@
+package com.katermar.assignment2;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Created by katermar on 5/23/2017.
+ */
+public class BinaryTree {
+    TreeNode root;
+
+    /*Inner class for Binary Tree node
+    * @field value stores integer value
+    * @field left, right store left and right subtrees*/
+    public static class TreeNode {
+        private final int value;
+        private TreeNode left;
+        private TreeNode right;
+
+        TreeNode(int item) {
+            value = item;
+        }
+
+        public int getValue() {
+            return value;
+        }
+
+        public TreeNode getLeft() {
+            return left;
+        }
+
+        @Override
+        public String toString() {
+            return String.format("TreeNode = %d", this.value);
+        }
+
+        public TreeNode getRight() {
+            return right;
+        }
+
+    }
+
+    /* Function to print all ancestors of purposed value
+    * @param node - root node
+    * @param value - value of the node on which we want to print ancestors
+    * @return true, if value exists in this tree, otherwise - false*/
+    boolean findAllAncestors(List<TreeNode> list, TreeNode node, int value)
+    {
+         //if tree is empty returns false
+        if (node == null) {
+            return false;
+        }
+
+        //if value exist in a tree
+        if (node.getValue() == value) {
+            return true;
+        }
+
+        //if value is present in left/right subtree
+        // then we'll print it
+        if (findAllAncestors(list, node.left, value) || findAllAncestors(list, node.right, value)) {
+            list.add(node);
+            return true;
+        }
+
+        //else return false
+        return false;
+    }
+
+    /*Traverse tree in preorder
+    * @param root - tree root*/
+    public void preorderTraversal (TreeNode root){ //метод для обхода в ширину
+        if (root != null){
+            System.out.print(root.getValue() + " ");
+            preorderTraversal (root.left);
+            preorderTraversal (root.right);
+        }
+    }
+
+    /*Populates tree with the array in order
+    * @param array - array which will fill the tree*/
+    public boolean populateTree(int[] array) {
+        for (int i = 0; i < array.length; i++) {
+            this.insert(array[i]);
+        }
+
+        return true;

```

Comment:
Since this method always returns true, it does not really make sense to have boolean as return type. Just void would be enough.